### PR TITLE
Checkbox

### PR DESCRIPTION
**Pull request by ktw1016**
- [x] Replace all `input type=checkbox` or any equivalents of checkboxes with this new `CheckBox` component (`AdvancedSearch, EstimatesComparison, simple_view, explorer_view`)
- [x] Make non-interactive `GraphLegend` solid checkboxes (`detailed_program_spending_split, goco, internal_services, SelectList, canada pov people graph`)
- [x] Replace `CheckboxSelector`
- [x] Replace `EmailFrontend`

let me know any feedback and if I missed any in the list above that I'm replacing

### CODE REVIEW

**Comment by ktw1016** [comment]
Could re-name `GraphLegend` to `CheckBoxContainer` or `CheckBoxList` or something like that since we're using it in rpb and TextDiff as well but probably not necessary

File: client/src/components/CheckBox.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    const {
+      id,
+      onClick,
+      color,
+      label,
+      active,
+      isSolidBox,
+      container_style,
+      checkbox_style,
+      label_style,
```
Comment (ktw1016) [inline client/src/components/CheckBox.js]: Not sure the best practices for these style props

**Comment by Stephen-ONeil** [comment]
Old style legend looks a bit busted up here:
image

**Comment by Stephen-ONeil** [comment]
Would it look good if the none-selected box has an x or a - in it, instead of being totally empty? Or would that be too visually noisy?

**Comment by ktw1016** [comment]
> Would it look good if the none-selected box has an x or a - in it, instead of being totally empty? Or would that be too visually noisy?

I personally think that would be confusing

File: client/src/components/CheckBox.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      checkMark_vertical_align,
+      disabled,
+    } = this.props;
```
[the remark this answers is not in the document]
Comment (ktw1016) [reply client/src/components/CheckBox.js]: ohhh yes

**Pull request merged**